src/persistence/index.js

```diff
@@ -76,6 +76,105 @@ export class PersistDB {
         return PersistDB.open().then(db => db.openDocument(id))
     }
 
+    /**
+     * Return JS Object of IndexedDB with DB_NAME.
+     */
+    static toJS() {
+        return PersistDB.open().then(async db => {
+            const tx = db.database.transaction(
+                ['states', 'changes', 'contents'], 'readonly')
+
+            const [states, changes, contents] = await Promise.all([
+                tx.objectStore('states').getAll(),
+                tx.objectStore('changes').index('document').getAll(),
+                tx.objectStore('contents').getAll(),
+            ].map(promisify))
+
+            return {
+                name: DB_NAME,
+                version: DB_VERSION,
+                states,
+                changes,
+                contents,
+            }
+        }).catch(e => {
+            throw new Error(e)
+        })
+    }
+
+    /**
+     * Replace database from JS Object. This will delete existing database and replace
+     * ot with new content.
+     *
+     * If database does not exists, it will be created.
+     *
+     * If you provide @param ids then this will remove only data for those ids.
+     *
+     * @param {Object} newDB
+     * @param {string[]} ids
+     */
+    static replaceDatabase(newDB = {}, ids = []) {
+        const {
+            name: newName,
+            version: newVersion,
+            states: newStates = [],
+            changes: newChanges = [],
+            contents: newContents = [],
+        } = newDB
+
+        if (newName !== DB_NAME) {
+            throw new Error(`Provided name "${newName}" does not match database name "${DB_NAME}".`)
+        }
+        if (Number(newVersion) !== DB_VERSION) {
+            throw new Error(`Provided version "${newVersion}" does not match database version "${DB_VERSION}."`)
+        }
+
+        PersistDB.open().then(async db => {
+            const tx = db.database.transaction(
+                ['states', 'changes', 'contents'], 'readwrite')
+            const states = tx.objectStore('states')
+            const changes = tx.objectStore('changes')
+            const contents = tx.objectStore('contents')
+
+            if (ids.length) {
+                let promises = []
+                ids.map(id => {
+                    promises.push(promisify(states.delete(id)))
+                    promises.push(promisify(contents.delete(id)))
+                })
+                await Promise.all([
```

Comment:
This could be simplified to `await Promise.all(ids.map(id => db.openDocument(id).then(d => d.discard())))`.

Reply:
When I'm doing this then I'm getting this error:
`Error: "TransactionInactiveError: A request was placed against a transaction which is currently not active, or which is finished."` for:
```
await Promise.all(statesToUpdate.map(st => 
    promisify(states.put(st))
))
```